codex/manifest/manifest.nim

```diff
@@ -308,5 +315,5 @@ proc new*(
     originalTreeCid: manifest.treeCid,
     originalDatasetSize: manifest.originalDatasetSize,
     verifiable: true,
-    verificationRoot: verificationRoot,
-    slotRoots: slotRoots))
+    slotsRoot: slotsRoot,
+    slotRoots: @slotRoots)
```

Comment:
Sure we want to copy here? It could be a `move` annotation in the future instead.

Reply:
Move would destroy the source tho? We should probably use sync/lent, but I want to leave that for a future pass.

Reply:
So, I've always assumed that `@` from `openArray` would _not_ make a copy, so I checked `system.nim` and...

```nim
  proc `@`*[T](a: openArray[T]): seq[T] {.magic: "OpenArrayToSeq".} =
    ## Turns an *openArray* into a sequence.
    ##
    ## This is not as efficient as turning a fixed length array into a sequence
    ## as it always copies every element of `a`.
    newSeq(result, a.len)
    for i in 0..a.len-1: result[i] = a[i]
```

Both, from the comments and the implementation, it does seem to make a copy... thats weird, why would it? But, it has a magic attached `OpenArrayToSeq`, so I checked the generated c source and...

```C
asgnRef((void**) (&(*T14_)._protected_0_1._verifiable_1.slotRoots), at___OOZOOZOOZcodexZslotsZbuilder_9449(slotRoots, slotRootsLen_0));
```

where `asgnRef` is 

```C
static N_INLINE(void, asgnRef)(void** dest, void* src) {
        ....
	nimln_(288, "/nim-codex/vendor/nimbus-build-system/vendor/Nim/lib/system/gc.nim");
	(*dest) = src;
	popFrame();
}
```

`(*dest) = src;` does a pointer assignment

So, it doesn't look like it's doing a copy.

Btw, a move wouldn't be appropriate here, since again, I'm simply converting `openArray` to a `seq`